include/zephyr/arch/xtensa/mpu.h

```diff
@@ -190,6 +190,85 @@ struct xtensa_mpu_map {
 	struct xtensa_mpu_entry entries[XTENSA_MPU_NUM_ENTRIES];
 };
 
+/**
+ * @name Memory domain and partitions
+ * @{
+ */
+
+typedef uint32_t k_mem_partition_attr_t;
+
+static inline bool xtensa_mem_partition_is_executable(k_mem_partition_attr_t access_rights)
```

Comment:
Isn't this all just duplicating logic from decode_permissions()?  I mean, same point obtains: I really don't think we want this bidirectional mapping and should be doing all our logic in "Zephyrese" and just translating in one way.  But if we have to do it, can we do it in only one place?

Reply:
Removed `decode_permission()`.